File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    dic = {}
    ssum = 0
    for i in range(len(name)):
        if name[i] not in dic:
            dic[name[i]] = yearning[i]
        ssum += yearning[i]
        
    for i in range(len(photo)):
        count = 0 
        no = 0
        for j in range(len(photo[i])):
            if photo[i][j] in dic:
                count += dic[photo[i][j]]
            
            
            
        answer.append(count)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    let obj = {}
    for(let i =0; i <name.length ; i++){
        obj[name[i]] = yearning[i];
    }
    
    for(let i =0; i < photo.length; i ++){
        let cnt = 0;
        for(let n of photo[i]){
            if(obj[n]){
                cnt += obj[n]
            }
        }
        answer.push(cnt)
    }
    return answer;
}